### DISCUSSION TITLE
Moving average channel code in Python?

### DISCUSSION THREAD

**Discussion by marudwarak** [DaveSkender/Stock.Indicators#327] [Q&A] [answered]
I need your assistance to create a moving average channel code for trading Index, DM me in telegram @marudwarak

**Comment by DaveSkender** [comment, marked as answer]
For use in Python, I've not tried it yet, but you can give [Python.NET](https://pypi.org/project/pythonnet) or comparable integrations a try.

There are many different moving average channel indicator types.  These are currently available in this library:

- [Moving Average Envelopes](https://daveskender.github.io/Stock.Indicators/indicators/MaEnvelopes/#content)
- [Bollinger Bands](https://daveskender.github.io/Stock.Indicators/indicators/BollingerBands/#content)
- [Donchian Channels](https://daveskender.github.io/Stock.Indicators/indicators/Donchian/#content)
- [Keltner Channels](https://daveskender.github.io/Stock.Indicators/indicators/Keltner/#content)
- [STARC Bands](https://daveskender.github.io/Stock.Indicators/indicators/StarcBands/#content)
- [SuperTrend](https://daveskender.github.io/Stock.Indicators/indicators/SuperTrend/#content)

Additionally, we also have many regular, non-channel style moving average types; currently:

- [Simple Moving Average (SMA)](https://daveskender.github.io/Stock.Indicators/indicators/Sma/#content)
- [Exponential Moving Average (EMA)](https://daveskender.github.io/Stock.Indicators/indicators/Ema/#content)
- [Arnaud Legoux Moving Average (ALMA)](https://daveskender.github.io/Stock.Indicators/indicators/Alma/#content)
- [Double Exponential Moving Average (DEMA)](https://daveskender.github.io/Stock.Indicators/indicators/Ema/#content)
- [Hull Moving Average (HMA)](https://daveskender.github.io/Stock.Indicators/indicators/Hma/#content)
- [Kaufman’s Adaptive Moving Average (KAMA)](https://daveskender.github.io/Stock.Indicators/indicators/Kama/#content)
- [MESA Adaptive Moving Averages (MAMA)](https://daveskender.github.io/Stock.Indicators/indicators/Mama/#content)
- [Triple Exponential Moving Average (TEMA)](https://daveskender.github.io/Stock.Indicators/indicators/Ema/#content)
- [Weighted Moving Average (WMA)](https://daveskender.github.io/Stock.Indicators/indicators/Wma/#content)

**Reply by DaveSkender** [reply]
As a follow up to this, we do now have a [moving average channel](https://daveskender.github.io/Stock.Indicators/indicators/MaEnvelopes/#content) indicator and are currently working on a Python wrapper.  See #394 to follow wrapper development, or to contribute.
